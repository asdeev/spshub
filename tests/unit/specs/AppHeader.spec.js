import { createLocalVue, shallowMount } from '@vue/test-utils';
import { expect } from 'chai';
import Router from 'vue-router';
import Vuetify from 'vuetify';
import testMap from '../../testMap';
import header from './fixtures/header';

const localVue = createLocalVue();
localVue.use(Router);
localVue.use(Vuetify);

describe('AppHeader', () => {
  let props;

  const build = () => {
    const wrapper = shallowMount(testMap.AppHeader, {
      localVue,
      propsData: props
    });

    return {
      wrapper,
      title: () => wrapper.find('.app-header__title'),
      contactButton: () => wrapper.find('.app-header__contact-button'),
      joinButton: () => wrapper.find('.app-header__join-button')
    };
  };

  beforeEach(() => {
    props = {
      header
    };
  });

  it('renders the component', () => {
    const { wrapper } = build();

    expect(wrapper.html()).to.matchSnapshot();
  });

  it('renders the main child components', () => {
    const { title, contactButton, joinButton } = build();

    expect(title().exists()).to.be.true;
    expect(contactButton().exists()).to.be.true;
    expect(joinButton().exists()).to.be.true;
  });
});
