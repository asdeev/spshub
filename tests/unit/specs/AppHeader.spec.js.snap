// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader renders the component 1`] = `"<v-card-stub tag=\\"div\\" tile=\\"true\\" flat=\\"true\\" class=\\"flex\\"><v-card-title-stub class=\\"app-header__title primary1\\"><div class=\\"title white--text\\">Somali Professionals &amp; Students Hub</div> <v-spacer-stub></v-spacer-stub> <v-btn-stub color=\\"info1 white--text\\" activeclass=\\"v-btn--active\\" tag=\\"button\\" type=\\"button\\" class=\\"app-header__contact-button\\">Contact Us</v-btn-stub> <v-btn-stub color=\\"info2 white--text\\" activeclass=\\"v-btn--active\\" tag=\\"button\\" type=\\"button\\" class=\\"app-header__join-button\\">Join SPSHub Today!&nbsp;<i class=\\"fas fa-chevron-right\\"></i></v-btn-stub></v-card-title-stub></v-card-stub>"`;
